torch/csrc/jit/codegen/cuda/runtime/array.cu

```diff
@@ -0,0 +1,130 @@
+// aligned register array for vectorized load/store
+template <typename scalar_t, int size, int align_size>
+struct alignas(sizeof(scalar_t) * align_size) Array {
+  scalar_t array[size];
+
+  __device__ void set(scalar_t v) {
+#pragma unroll
+    for (int i = 0; i < size; ++i) {
+      array[i] = v;
+    }
+  }
+
+  __device__ scalar_t& operator[](const unsigned int i) {
+    return array[i];
+  }
+};
+
+// Used for vectorized allocations that are not in registers
+template <typename scalar_t, int vec_size>
+void arraySet(scalar_t* buff, scalar_t val) {
```

Comment:
Would we need specializations for vectorized initialization? We could also rely on compiler's auto-vectorization pass.

Reply:
I didn't explicitly check as it is unlikely to be a perf bottleneck for memory bound ops. If it's necessary we could definitely do it.